catalyst/contrib/dl/callbacks/inference.py

```diff
@@ -0,0 +1,213 @@
+from typing import List, Tuple, Union
+from pathlib import Path
+
+import numpy as np
+
+import torch
+import torch.nn.functional as F
+
+from catalyst.contrib.data.cv.datasets import get_image_margins
+from catalyst.core import Callback, CallbackOrder, State
+
+
+def pyramid_weights(height: int, width: int):
+    """
+    Computes a weight matrix that assigns bigger weights
+    on pixels at center and less weight to pixels on image boundary.
+    This weight matrix then used for merging individual tile predictions
+    and helps dealing with prediction artifacts on tile boundaries.
+
+    Args:
+        height: Tile height
+        width: Tile width
+
+    Returns:
+        Weight matrix of shape [`height` x `width`]
+    """
+    x_center = width * 0.5
+    y_center = height * 0.5
+    x_left = 0
+    x_right = width
+    y_top = 0
+    y_bottom = height
+
+    d_center_x = torch.pow(torch.arange(width) - x_center + 0.5, 2)
+    d_center_y = torch.pow(torch.arange(height) - y_center + 0.5, 2)
+    d_center = torch.sqrt(d_center_y.unsqueeze(0).transpose(0, 1) + d_center_x)
+
+    d_edge_left = torch.pow(torch.arange(width) - x_left + 0.5, 2)
+    d_edge_right = torch.pow(torch.arange(width) - x_right + 0.5, 2)
+    d_edge_top = torch.pow(torch.arange(height) - y_top + 0.5, 2)
+    d_edge_bottom = torch.pow(torch.arange(height) - y_bottom + 0.5, 2)
+
+    d_edge_x = torch.sqrt(torch.min(d_edge_left, d_edge_right) + 0.5 ** 2)
+    d_edge_y = torch.sqrt(torch.min(d_edge_top, d_edge_bottom) + 0.5 ** 2)
+    d_edge = torch.min(d_edge_y.unsqueeze(0).transpose(0, 1), d_edge_x)
+
+    frac = torch.div(d_edge, d_center + d_edge)
+    alpha = (width * height) / torch.sum(frac).item()
+    weights = alpha * frac
+
+    return weights
+
+
+class TiledInferenceCallback(Callback):
```

Comment:
minor: probably it would be better to rename this callback to `TiledDumpingCallback` or something like this.

Reply:
Question: Why do we need to store all tiles and predictions?

Reply:
What means for dumping? It it inference-only feature, leading to its name.

Reply:
> minor: probably it would be better to rename this callback to `TiledDumpingCallback` or something like this.

What means for dumping? It it inference-only feature, leading to its name.

Reply:
> Question: Why do we need to store all tiles and predictions?

To form resulting mask. I don't understand your question :)